src/main/java/com/thealgorithms/datastructures/stacks/NodeStack.java

```diff
@@ -3,159 +3,65 @@
-     * @value next : the next value from this node
-     * @value previous : the last value from this node
-     * @value size : size of the stack
-     */
     private Item data;
+    private NodeStack<Item> previous;
+    private NodeStack<Item> head;
+    private int size = 0;
 
-    private static NodeStack<?> head;
-    private NodeStack<?> previous;
-    private static int size = 0;
-
-    /**
-     * Constructors for the NodeStack.
-     */
     public NodeStack() {
     }
 
     private NodeStack(Item item) {
         this.data = item;
     }
 
-    /**
-     * Put a value onto the stack.
-     *
-     * @param item : value to be put on the stack.
-     */
     public void push(Item item) {
-        NodeStack<Item> newNs = new NodeStack<Item>(item);
+        NodeStack<Item> newNode = new NodeStack<>(item);
 
-        if (this.isEmpty()) {
-            NodeStack.setHead(new NodeStack<>(item));
-            newNs.setNext(null);
-            newNs.setPrevious(null);
+        if (isEmpty()) {
+            head = newNode;
         } else {
-            newNs.setPrevious(NodeStack.head);
-            NodeStack.head.setNext(newNs);
-            NodeStack.setHead(newNs);
+            newNode.previous = head;
+            head = newNode;
         }
-
-        NodeStack.setSize(NodeStack.getSize() + 1);
+        size++;
     }
 
-    /**
-     * Value to be taken off the stack.
-     *
-     * @return item : value that is returned.
-     */
     public Item pop() {
-        Item item = (Item) NodeStack.head.getData();
-
-        NodeStack.setHead(NodeStack.head.getPrevious());
-        NodeStack.head.setNext(null);
-
-        NodeStack.setSize(NodeStack.getSize() - 1);
+        if (isEmpty()) {
+            throw new IllegalStateException("Stack is empty, cannot peek element");
+        }
 
+        Item item = head.data;
+        head = head.previous;
+        size--;
         return item;
     }
 
-    /**
-     * Value that is next to be taken off the stack.
-     *
-     * @return item : the next value that would be popped off the stack.
-     */
     public Item peek() {
-        return (Item) NodeStack.head.getData();
+        if (isEmpty()) {
+            throw new IllegalStateException("Stack is empty, cannot peek element");
+        }
+        return head.data;
     }
 
-    /**
-     * If the stack is empty or there is a value in.
-     *
-     * @return boolean : whether or not the stack has anything in it.
-     */
     public boolean isEmpty() {
-        return NodeStack.getSize() == 0;
+        return size == 0;
     }
 
-    /**
-     * Returns the size of the stack.
-     *
-     * @return int : number of values in the stack.
-     */
     public int size() {
-        return NodeStack.getSize();
+        return size;
     }
 
-    /**
-     * Print the contents of the stack in the following format.
-     *
-     * <p>
-     * x <- head (next out) y z <- tail (first in) . . .
-     */
     public void print() {
```

Comment:
Please remove print function, we don't provide visualization means, only the data structure itself

Reply:
ok no problem...sorry

Reply:
It's still here